tests/js/zombies/zombie-forgot.js

```diff
@@ -0,0 +1,287 @@
+                .pressButton("Send Email", function () {
+                    jqUnit.start();
+
+                    // The "forgot password" form should be visible
+                    var forgotForm = browser.window.$(".forgot-form");
+                    jqUnit.assertNotUndefined("There should be a \"forgot password\" form...", forgotForm.html());
+                    jqUnit.assertEquals("The \"forgot password\" form should not be hidden...", "", forgotForm.css("display"));
+
+                    // A "success" message should be visible
+                    var feedback = browser.window.$(".success");
+                    jqUnit.assertUndefined("There should not be a positive feedback message...", feedback.html());
+
+                    // There should be no alerts
+                    var alert = browser.window.$(".alert");
+                    jqUnit.assertNotUndefined("There should be an alert...", alert.html());
+                    if (alert.html()) {
+                        jqUnit.assertTrue("The alert should have content.", alert.html().trim().length > 0);
+                    }
+                });
+        });
+    });
+
+    jqUnit.asyncTest("Try to use an invalid reset code...", function () {
+        var timestamp = (new Date()).getTime();
+        browser.visit(harness.options.baseUrl + "content/reset?code=" + timestamp).then(function () {
+            jqUnit.start();
+            isBrowserSane(jqUnit, browser);
+            jqUnit.stop();
+
+            browser.fill("password", timestamp)
+                .fill("confirm", timestamp)
+                .pressButton("Reset Password", function () {
+                    jqUnit.start();
+
+                    // The "forgot password" form should not be visible
+                    var resetForm = browser.window.$(".reset-form");
+                    jqUnit.assertNotUndefined("There should be a \"reset\" form...", resetForm.html());
+
+                    // A "success" message should not be visible
+                    var feedback = browser.window.$(".success");
+                    jqUnit.assertUndefined("There should not be a positive feedback message...", feedback.html());
+
+                    // There should be at least one alert
+                    var alert = browser.window.$(".alert");
+                    jqUnit.assertNotUndefined("There should be an alert...", alert.html());
+                    if (alert.html()) {
+                        jqUnit.assertTrue("The alert should have content.", alert.html().trim().length > 0);
+                    }
+                });
+        });
+    });
+
+    // Save the test that uses a mail handler for the last.
+    //
+    // If we have to add even one more test that works with email, we will need to refactor to use a testEnvironment
+    // and testCases, as we cannot reuse a mail handler between tests.
+    jqUnit.asyncTest("Reset a user's password using the \"forgot password\" form...", function () {
```

Comment:
@amb26 I could use your help in talking through wrapping key portions of Zombie in components that can be more easily wired into test sequences.